### ISSUE TITLE
torch lr scheduler 

### ISSUE THREAD

**Issue by luome** [epfml/powersgd#19]
hi, powersgd is good in my practice, In which way can powersgd work with torch lr scheduler?

**Comment by tvogels** [comment]
Hi @luome 

You can see PowerSGD as a way of aggregating gradients across nodes, rather than an optimizer. You can combine this aggregator with any pytorch optimizer you like.

If you're using the code from the main branch of this repository, you can use custom learning rate schedules by customizing this function:

https://github.com/epfml/powersgd/blob/master/powersgd/__init__.py#L21 

While this is the simplest way to do it, you might also want to try to scale the gradients *before* aggregating them with PowerSGD. This is not completely equivalent, and may work better.

You can also use the [PyTorch DDP implementation](https://pytorch.org/docs/stable/ddp_comm_hooks.html#torch.distributed.algorithms.ddp_comm_hooks.powerSGD_hook.PowerSGDState) of PowerSGD. That code is stable and maintained by the PyTorch team, but might be more difficult to customize than this codebase.

Hope this helps. Let me know if you have further questions.

**Comment by luome** [comment]
Thanks a lot!
